poetry/flatpak-poetry-generator.py

```diff
@@ -63,19 +63,23 @@ def get_module_sources(parsed_lockfile: dict, include_devel: bool = True) -> lis
     for section, packages in parsed_lockfile.items():
         if section == "package":
             for package in packages:
-                if (
-                    package["category"] == "dev"
-                    and include_devel
-                    and not package["optional"]
-                    or package["category"] == "main"
-                    and not package["optional"]
+                if "category" not in package or (
+                    (
+                        package.get("category") == "dev"
+                        and include_devel
+                        and not package.get("optional")
+                    )
+                    or (
+                        package.get("category") == "main"
+                        and not package.get("optional")
+                    )
                 ):
```

Comment:
the original logic and the rewritten one can be simplified a bit I think. It's a bit hard to follow this many ands/ors. Something like

```
category = package.get("category")
optional = package.get("optional", False)

if (
    not category
    or (category == "dev" and include_devel and not optional)
    or (category == "main" and not optional)
):
```

Reply:
Applied